meson.build

```diff
@@ -0,0 +1,95 @@
+project('fluidsynth', 'c',
+        version : '2.1.7',
+        license : 'LGPL-2.1-or-later',
+        default_options : ['b_ndebug=if-release'],
+        meson_version : '>= 0.53.0')
+
+cc = meson.get_compiler('c')
+
+summary('Build type', get_option('buildtype'), section : 'Build Summary')
+
+# config file
+#
+# TODO there are many, many more options to set/detect in upstream template
+#      see meson_options.txt for list
+
+conf = configuration_data()
+
+if host_machine.system() == 'darwin'
+  conf.set('DARWIN', 1)
+endif
+
+conf.set('DEFAULT_SOUNDFONT', '"/usr/local/share/soundfonts/default.sf2"')
+
+conf.set10('STDC_HEADERS', true)
+
+if get_option('enable-floats')
+  conf.set('WITH_FLOAT', 1)
+endif
+summary('Samples type', conf.has('WITH_FLOAT') ? 'float' : 'double',
+        section : 'Developer nerds info')
+
+headers_to_check = [
+  'arpa/inet.h',        'errno.h',            'fcntl.h',
+  'getopt.h',           'inttypes.h',         'io.h',
+  'limits.h',           'linux/soundcard.h',  'machine/soundcard.h',
+  'math.h',             'netinet/in.h',       'netinet/tcp.h',
+  'pthread.h',          'signal.h',           'stdarg.h',
+  'stdint.h',           'stdio.h',            'stdlib.h',
+  'string.h',           'strings.h',          'sys/mman.h',
+  'sys/socket.h',       'sys/soundcard.h',    'sys/stat.h',
+  'sys/time.h',         'sys/types.h',        'unistd.h',
+  'windows.h',
+]
+
+foreach h : headers_to_check 
+  conf.set10('HAVE_' + h.underscorify().to_upper(), cc.has_header(h))
+endforeach
+
+if cc.has_function('inet_ntop', prefix : '#include <arpa/inet.h>')
+  conf.set('HAVE_INETNTOP', 1) # lib uses ifdef for this one
+endif
+
+foreach fun : ['sinf', 'cosf', 'fabsf', 'powf', 'sqrtf', 'logf']
+  conf.set10('HAVE_' + fun.to_upper(), cc.has_function(fun))
+endforeach
+
+
+# external dependencies
+
+m_dep       = cc.find_library('m', required : false)
+glib_dep    = dependency('glib-2.0', static : true)
+gthread_dep = dependency('gthread-2.0', static : true)
```

Comment:
Why are these static?

Reply:
In one sentence: to prevent generating broken AppBundles on macOS. But I was a bit lazy with hardcoding this to `true`, in dosbox-staging I dealt with this by adding a user option, and explaining in more detail: https://github.com/dosbox-staging/dosbox-staging/blob/master/meson_options.txt#L59-L72

As it happens, on macOS, when passing `static:true` to `glib`, `gthread`, and other libs, like e.g. `SDL2` - Meson (correctly) warns that this might not work because `iconv` and `objc` can't be linked statically - but those two libs are part of macOS system libraries, so resulting binary is works as it should. We then can just place it in macOS AppBundle / directory and distribute - users don't need to have the dependencies installed via `brew`.

If I am missing some switch to accomplish the same using built-in Meson switches, that would be better - but I couldn't find anything - `--default-library` affects only libraries built (either in project or in wraps), not the `static` kwarg of dependency object.

Reply:
I've done the same thing here: https://git.archlinux.org/pacman.git/tree/meson_options.txt#n4

`-Dbuildstatic=true -Ddefault_library=static` -- the first guarantees all dependencies are static, the second guarantees all internal libs are static.

But currently the OOTB experience for this wrap produces:

```
Found pkg-config: /usr/bin/pkg-config (1.7.3)
WARNING: Static library 'glib-2.0' not found for dependency 'glib-2.0', may not be statically linked
WARNING: Static library 'pcre' not found for dependency 'glib-2.0', may not be statically linked
Run-time dependency glib-2.0 found: YES 2.66.7
WARNING: Static library 'gthread-2.0' not found for dependency 'gthread-2.0', may not be statically linked
WARNING: Static library 'glib-2.0' not found for dependency 'gthread-2.0', may not be statically linked
WARNING: Static library 'pcre' not found for dependency 'gthread-2.0', may not be statically linked
Run-time dependency gthread-2.0 found: YES 2.66.7
```

So I think adding a meson_options.txt selection defaulting to false is the correct approach here.

Reply:
I named the option `try-static-deps` to signalize to the end-user, that build might succeed without these libraries being statically linked. Do you think it would be beneficial to have a "blessed" name for such option inside Meson?

Reply:
Thanks, lgtm.

I think there could be some benefit to having a reserved core option in meson, rather than a project option. This is covered by the following outstanding feature request: https://github.com/mesonbuild/meson/issues/7621